Source/Core/Engine/ResourceSystem.cpp

```diff
@@ -169,6 +169,12 @@ ResourceLibrary* ResourceSystem::LoadPackage(Status& status, ResourcePackage* pa
   resourceLibrary->Name = package->Name;
   LoadedResourceLibraries.InsertOrError(package->Name, resourceLibrary);
 
+  if (PL::gContentSystem->PlasmaCoreLibraryNames.Contains(package->Name))
+  {
+      LoadedPlasmaCoreLibraries.InsertOrError(package->Name, resourceLibrary);
+      LoadedDependencyLibraries.InsertOrError(package->Name, resourceLibrary);
+  }
+
   if (lastResourceLibrary != nullptr)
     resourceLibrary->AddDependency(lastResourceLibrary);
```

Comment:
This is because it functions recursively for some reason, which I myself am not a big fan of. It's a linked list which is *okay* but dependencies should be changed to be just a straight list instead, but that would require way more time on this and this is more of a bandaid fix instead of a permanent fix.

We are going to have to have to overhaul most of the library loading system to accommodate this.